packages/ui/components/ContractView/TreasuryView.tsx

```diff
@@ -44,6 +45,25 @@ export const TreasuryBalances: FC<TreasuryBalancesProps> = ({
       {nativeTokens.map(({ denom, amount, decimals }) => {
         const symbol = nativeTokenLabel(denom)
         const icon = nativeTokenLogoURI(denom)
+        if (symbol.startsWith('IBC')) {
+          // We're dealing with an IBC token we don't know about. Instead
+          // of showing a long hash, hide that in a tooltip.
+          return (
+            <BalanceListItem key={symbol}>
+              <UnknownAssetBalanceIcon />
+              {convertMicroDenomToDenomWithDecimals(
+                amount,
+                decimals
+              ).toLocaleString(undefined, {
+                maximumFractionDigits: decimals,
+              })}{' '}
+              <span className="flex flex-row gap-1">
+                <p>Unknown Asset</p>
```

Comment:
Not sure why the linter passed. Would you kindly move this to the en translation (probably `info.unknownAsset`) :D